### ISSUE TITLE
[clang-format] "AllowShortFunctionsOnASingleLine: Inline" doesn't works after a "#pragma" directive

### ISSUE THREAD

**Issue by aarcangeli** [llvm/llvm-project#55260]
When I upgraded clang-format to version 14, I noticed that with configuration `AllowShortFunctionsOnASingleLine` set to `Inline` or `InlineOnly` many methods are actually formatted as if it were set to `None`.

It seems that when a pragma directive is used, subsequent statements are treated as "out of class".

This was working correctly with clang-format 13. Is it expected?

### Steps to reproduce

1. With the following files in WD:

```cpp
// test.h
class Ciao {
  void foo1(float value)
  { int t = 0; }
#pragma TestRegion
  void foo2(float value)
  { int t = 0; }
};
```

```yml
# .clang-format
Language: Cpp
BasedOnStyle: LLVM
AllowShortFunctionsOnASingleLine: InlineOnly
```

2. Reformat file using `clang-format -i test.h`

**expected result**

```cpp
// test.h
class Ciao {
  void foo1(float value) { int t = 0; }
#pragma TestRegion
  void foo2(float value) { int t = 0; }
};
```

**actual result**

```cpp
// test.h
class Ciao {
  void foo1(float value) { int t = 0; }
#pragma TestRegion
  void foo2(float value) {
    int t = 0;
  }
};
```

### Reference

https://clang.llvm.org/docs/ClangFormatStyleOptions.html#:~:text=configuration%3A%20InlineOnly)-,Only%20merge%20functions%20defined%20inside%20a%20class,-.%20Same%20as%20%E2%80%9Cinline

### Environment
```
clang-format --version
clang-format version 14.0.3 (https://github.com/ssciwr/clang-format-wheel 5ccf159e31550ec77a1e7e1e3ae31bb04c3d0b90)
```

**Comment by aarcangeli** [comment]
- I changed the issue to make it more clear
- Bump version `14.0.3`

**Comment by lukash** [comment]
+1. Seems to happen after any preprocessor directive inside the class.

**Comment by m-blaha** [comment]
It's been fixed in version 15.
```
# clang-format --version
clang-format version 15.0.4 (Fedora 15.0.4-1.fc37)
```

**Comment by rymiel** [comment]
This was fixed by f14ebe91c5dd6be5b64a45e479291cd08676be0c
